src/security/ServerOptions.cc

```diff
@@ -432,18 +432,21 @@ Security::ServerOptions::updateContextClientCa(Security::ContextPointer &ctx)
 
         if (parsedFlags & SSL_FLAG_DELAYED_AUTH) {
             debugs(83, 9, "Not requesting client certificates until acl processing requires one");
-            SSL_CTX_set_verify(ctx.get(), SSL_VERIFY_NONE, nullptr);
+            Ssl::SetupPeerVerification(ctx, false, false);
+        } else if(parsedFlags & SSL_FLAG_CONDITIONAL_AUTH) {
+            debugs(83, 9, "Allowing client certificates.");
```

Comment:
A few polishing touches here:

1. I have not checked, but I suspect that client certificates are, technically, always "allowed" -- the server cannot stop the client from supplying more information. Let's see if we can improve this wording. I have a specific suggestion below.
2. Please use debugs() level 5 here, despite the inconsistency wit similar debugs() statements nearby. Level 9 is reserved for data dumps in new code. 
3. Also, please do not format routine low-level debugging as a sentence, for brevity and consistency sake (again, the other nearby debugs() statements are wrong, and it is overall better not to mimic them here).

```suggestion
            debugs(83, 5, "will request the client certificate but ignore its absence");
```